gui/app/services/updatesService.js

```diff
@@ -0,0 +1,89 @@
+(function(){
+  
+ //This handles updates
+ 
+ const _ = require('underscore')._;
+ const JsonDb = require('node-json-db');
+ const request = require('request');
+ const compareVersions = require('compare-versions');
+
+ angular
+  .module('firebotApp')
+  .factory('updatesService', function (settingsService) {
+    // factory/service object
+    var service = {}
+
+    // Update Checker
+    // This checks for updates.
+    service.updateCheck = function(){
+        return new Promise((resolve, reject) => {
```

Comment:
So, I tried this change in the service file and couldn't get it to play nice. Wrapping the entire promise in $q.when seems to break things, so I tried wrapping only the resolve and that made no difference.

So, I moved the $q.when to the controller and just wrapped the function call. That seems to have done the trick and I was able to remove scope.apply.